### DISCUSSION TITLE
What is a lazy bookshelf and what does it do?

### DISCUSSION THREAD

**Discussion by scorpio862** [advplyr/audiobookshelf#221] [General]
This was rescently added to the server and android app and I'm just wondering what does it do :)

**Comment by advplyr** [comment]
The original bookshelf was loading ALL of the books when the page loaded. The more books you had the slower the page would load.

The lazy bookshelf will fetch and mount books as they are needed (as you scroll down the page), and will remove books from the page when they are out of view. "Lazy" comes from [lazy loading](https://developer.mozilla.org/en-US/docs/Web/Performance/Lazy_loading).

This is also implemented on the mobile app now.

**Reply by scorpio862** [reply]
Thanks for explaining. It did speed up the library loading times. 

I think you could speed up the page load times evern more if all the high resolution covers will be automatically downsized to 208x332px (or whatever is the highest image size used by audiobookshelf).
I just noticed that some of my covers in the medatadata folder are more than 2MB. I have 425 audiobooks and all the covers in total have around 19MB and this is a lot of loading.

**Comment by Jdiesel87** [comment]
Slightly off topic but is there any dynamic image scaling happening with covers or is it displaying covers at their native resolution? I think other projects I have seen used ImageMagick for transforming images.

**Reply by advplyr** [reply]
I first responded logged into another account, if you are wondering why the previous message is gone.

@scorpio862 opened a ticket for this #223 and I put down some ideas. I think we can use ffmpeg to scale the images since we are already using it for audio transcoding.
But yeah, this will be a good performance improvement when implemented.

**Reply by keaganhilliard** [reply]
I created a quick PR #233 with an endpoint that dynamically sizes images based on query params. That's generally how the other streaming apps like emby and Plex handle this. I used a node library focused on image manipulation rather than ffmpeg, but it seems to work pretty well.

**Reply by advplyr** [reply]
In `v1.6.40` covers are optimized and cached, see #223.
This version doesn't include an arm7 build, hopefully that gets resolved in the next release.
